Add blob support

macintux opened the pull request:
See https://github.com/basho/riak_kv/pull/1540 for details.

macintux committed:
Print blob data type as hex

Rather than interpret binary data as a garbage string, display it as a
hex integer.

macintux committed:
Drop extraneous line feed

andytill commented on src/connection_srv.erl:
```diff
@@ -77,14 +77,30 @@ init([ShellRef, Nodes]) ->
-map_timestamp({[], {Name, timestamp}}) ->
+%% Useful for outputting blob types
+truncate_hex(Str, Len) when length(Str) > Len ->
+    string:substr(Str, 1, Len-3) ++ "...";
```
@gordonguthrie what do you think about truncation? I know it might be useful in the terminal, but what about shell logging that is later used for test verification?

andytill replied:
We discussed this today and decided that we should split blobs across several lines to increase the size of blob we can sensibly show, but we still need to truncate for very large values.

andytill commented on src/connection_srv.erl:
```diff
@@ -77,14 +77,30 @@ init([ShellRef, Nodes]) ->
+truncate_hex(Str, _Len) ->
+    Str.
+
+hex_as_string(Bin) ->
```
The `mochihex` module is used in riak_ql to convert between binary and hex.

macintux replied:
Updated, thanks

macintux committed:
Use mochihex

Andy correctly points out that the `hexlify` code exists in a library on
which we already depend, so use it.

macintux commented:
@andytill Need your final review for this and https://github.com/basho/riak_test/pull/1212

andytill commented:
+1

andytill commented:
Internet is so bad on this train I gave up for a while.

macintux commented:
Thanks!

gordonguthrie commented:
+1

hmmr commented:
+1
